src/Components/Endpoints/src/Discovery/PageCollectionBuilder.cs

```diff
@@ -0,0 +1,153 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Buffers;
+
+namespace Microsoft.AspNetCore.Components;
+
+/// <summary>
+/// Represents the list of pages in a <see cref="ComponentApplicationBuilder"/>.
+/// </summary>
+internal class PageCollectionBuilder
+{
+    private readonly List<PageComponentBuilder> _pages = new();
```

Comment:
Is it worth making this a dictionary so all the O(N) and O(N*2) operations below can be a factor of N cheaper?

Reply:
switched to a dictionary